### PR TITLE
WSL: Add support for moby backend.

### PR DESCRIPTION
**Pull request by mook-as**
This sets things up so that we can run moby on Windows.
Additionally, we add a proxy process so that `docker run` etc. works _from Windows_.

We'll need additional changes for integration with other Linux distributions.

---
Notes:
- We're starting `dockerd` on the WSL Linux side using OpenRC dependencies; no need to manually handle an extra process.
- The OpenRC scripts are now named `*.initd` to match what alpine-lima is doing.

### CODE REVIEW

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Mainly the typo and the use of `cfg.containerEngine` should be changed.

File: src/k8s-engine/wsl.ts [hunk 1]
```diff
@@ -840,16 +937,21 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
         ]);
 
         await this.startService('k3s', {
-          PORT:          this.#desiredPort.toString(),
-          LOG_DIR:       await this.wslify(paths.logs),
-          IPTABLES_MODE: 'legacy',
+          PORT:                   this.#desiredPort.toString(),
+          LOG_DIR:                await this.wslify(paths.logs),
+          'export IPTABLES_MODE': 'legacy',
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/k8s-engine/wsl.ts:942]: Spawned #1015 for this.

File: src/k8s-engine/lima.ts [hunk 2]
```diff
@@ -900,7 +900,7 @@ ${ commands.join('\n') }
     await this.writeFile('/etc/init.d/k3s', SERVICE_K3S_SCRIPT, 0o755);
     await this.writeConf('k3s', {
       PORT:   this.desiredPort.toString(),
-      ENGINE: this.#currentContainerEngine === ContainerEngine.MOBY ? '--docker' : '',
+      ENGINE: this.cfg?.containerEngine ?? ContainerEngine.CONTAINERD,
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/k8s-engine/lima.ts]: Fixed in e2aa41f8802afd70b80f67afee6f6ea167e64bc6.

File: src/k8s-engine/wsl.ts [hunk 3]
```diff
@@ -89,6 +167,14 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
       this.progress = progress;
       this.emit('progress');
     });
+    this.mobySocketProxyProcesss = new BackgroundProcess(this, async() => {
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/k8s-engine/wsl.ts]: My precioussss… fixed in e2aa41f8802afd70b80f67afee6f6ea167e64bc6.

**Review by ericpromislow — APPROVED** [review]
LGTM

**Pull request merged**